File: eval-reviews/2026-09-12-chained/workflow-recovery/python/opus/r1/starter/main.py
"""One-request JSON adapter; each request owns a fresh simulator."""
import json
import sys
from workflow import DomainError, build


def main() -> None:
    try:
        request = json.load(sys.stdin)
        result = build(request["input"]).run()
        response = {"ok": True, "result": result}
    except DomainError as error:
        response = {"ok": False, "error": {"code": error.code}}
    except (ValueError, KeyError, TypeError):
        response = {"ok": False, "error": {"code": "INVALID_INPUT"}}
    print(json.dumps(response, separators=(",", ":")))


if __name__ == "__main__":
    main()

File: eval-reviews/2026-09-12-chained/workflow-recovery/typescript/opus/r1/starter/main.ts
import { readFileSync } from "node:fs";
import { DomainError, evaluateRequest } from "./workflow.ts";

let response: unknown;
try {
  const request = JSON.parse(
    readFileSync(0, "utf8"),
    (_key: string, value: unknown, context?: { source?: string }) => {
      // Preserve the protocol's integer-token rule before JSON's numeric coercion loses it.
      if (
        typeof value === "number" &&
        context?.source &&
        !/^-?(0|[1-9][0-9]*)$/.test(context.source)
      )
        throw new DomainError("INVALID_INPUT");
      return value;
    },
  ) as { input?: unknown };
  response = { ok: true, result: evaluateRequest(request?.input) };
} catch (error) {
  if (error instanceof DomainError)
    response = { ok: false, error: { code: error.code } };
  else if (error instanceof SyntaxError)
    response = { ok: false, error: { code: "INVALID_INPUT" } };
  else throw error;
}
process.stdout.write(JSON.stringify(response) + "\n");
